### PR TITLE
Web3 1083 ci build and publish

### PR DESCRIPTION
**Pull request by rushby**
Parallel test runs, increase test coverage, add CI pipelines and slack notifications.

### CODE REVIEW

File: src/api/start/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+        mockOptions = {
+            host: SupportedNetwork.Testnet,
+        };
+
+        await expect(startSession(mockOptions)).rejects.toThrow(
+            'startSession should not be called in a browser environment, use "startWalletSession"',
+        );
+
+        delete (global as any).window;
```
Comment (yurikoinaba) [inline src/api/start/index.test.ts:86]: I think this is redundant bc it is in the `afterEach`

File: src/api/start/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    });
+
+    afterEach(() => {
+        delete (global as any).window;
+    });
+
+    it('should throw an error when called outside of a browser environment', async () => {
+        delete (global as any).window;
```
Comment (yurikoinaba) [inline src/api/start/index.test.ts:134]: Same here

File: tests/common/utils.ts [hunk 3]
```diff
@@ -20,6 +20,28 @@ export interface ProofData {
...
+export const proofTypes = Object.keys(ProofType).map((key) => ProofType[key as keyof typeof ProofType]);
+export const curveTypes = Object.keys(Groth16CurveType).map((key) => Groth16CurveType[key as keyof typeof Groth16CurveType]);
+
+const seedPhrases = [
+    process.env.SEED_PHRASE_1,
+    process.env.SEED_PHRASE_2,
+    process.env.SEED_PHRASE_3,
+    process.env.SEED_PHRASE_4,
+    process.env.SEED_PHRASE_5,
+    process.env.SEED_PHRASE_6,
```
Comment (yurikoinaba) [inline tests/common/utils.ts:32]: I think this is missing `process.env.SEED_PHRASE_7`

Comment (rushby) [reply tests/common/utils.ts:32]: 7 is more of a generic one used in the unit tests whereas 1 - 6 are for the proof types / variants.  Will add this as explanation.

**Review by jmontesdeocanuez — APPROVED** [review]

**Pull request merged**